src/robusta/core/sinks/pagerduty/pagerduty_sink.py

```
@@ -0,0 +1,124 @@
+        custom_details["Resource"] = finding.subject.name
+        custom_details["Cluster running Robusta"] = self.cluster_name
+        custom_details["Namespace"] = finding.subject.namespace
+        custom_details["Node"] = finding.subject.node
+        custom_details["Source of the Alert"] = str(finding.source.name)
+        custom_details["Severity"] = PagerdutySink.__to_pagerduty_severity_type(finding.severity).upper()
+        custom_details["Fingerprint ID"] = finding.fingerprint
+        custom_details["Description"] = finding.description
+        custom_details["Caption"] = f"{finding.severity.to_emoji()} {PagerdutySink.__to_pagerduty_severity_type(finding.severity)} - {finding.title}"
+
+        message_lines = ""
+        if finding.description:
+            message_lines = finding.description + "\n\n"
+
+        for enrichment in finding.enrichments:
+            for block in enrichment.blocks:
+                text = self.__to_unformatted_text(block)
+                if not text:
+                    continue
+
+                message_lines += text + "\n\n"
+
+        custom_details["state_message"] = message_lines                
+
+        body = {
+            "payload": {
+                "summary": finding.title,        
+                "severity": PagerdutySink.__to_pagerduty_severity_type(finding.severity),
+                "source": self.cluster_name,
+                "component": str(finding.subject),
+                "group":finding.service_key,
+                "class": finding.aggregation_key,
+                "custom_details": custom_details,
+            },
+            "routing_key": self.api_key,            
+            "event_action": PagerdutySink.__to_pagerduty_status_type(finding.title),
+            "dedup_key": finding.fingerprint
+        }
+        
+        headers = {"Content-Type": "application/json"}
+        response = requests.post(self.url, json=body, headers=headers)           
```

I think we should log an error here. When testing I put in an incorrect API key and it silently fails without notifying the user.

You can do something like this:
```
 if response.status_code != 200:
   logging.error(f"Error sending message to PagerDuty: {response.status_code}, {response.reason}, {response.text}")
```

---

Whoops, my bad. You should check `response.ok` and not explicitly check for a status_code of 200. PagerDuty actually returns 202 errors, which are also OK.